ui/app/src/components/Header/SearchBar.tsx

```diff
@@ -0,0 +1,223 @@
+                  pre: '<strong style="color:darkorange">',
+                  post: '</strong>',
+                  escapeHTML: true,
+                }).metadata.name,
+              }}
+            />
+          </Box>
+        </Button>
+      ))}
+    </Box>
+  );
+}
+
+function SearchDashboardList(props: { list: DashboardResource[]; query: string; onClick: () => void }) {
+  const kvSearch = useMemo(() => new KVSearch<DashboardResource>(kvSearchConfig), []);
+
+  const filteredList: Array<KVSearchResult<DashboardResource>> = useMemo(() => {
+    if (props.query) {
+      return kvSearch.filter(props.query, props.list);
+    } else {
+      return [];
+    }
+  }, [kvSearch, props.list, props.query]);
+
+  return filteredList.length === 0 ? (
+    <></>
+  ) : (
+    <Box sx={{ display: 'flex', flexDirection: 'column' }}>
+      <Box
+        sx={{
+          display: 'flex',
+          flexDirection: 'row',
+          alignItems: 'center',
+          justifyContent: 'flex-start',
+          marginBottom: 1,
+          marginLeft: 0.5,
+        }}
+      >
+        <ViewDashboardIcon sx={{ marginRight: 0.5 }} fontSize="medium" />
+        <Typography variant="h3">Dashboards</Typography>
+      </Box>
+      {filteredList.slice(0, sizeList).map((search) => (
+        <Button
+          variant="outlined"
+          sx={{
+            display: 'flex',
+            justifyContent: 'space-between',
+            marginBottom: 1,
+            marginLeft: 1,
+            marginRight: 1,
+          }}
+          component={RouterLink}
+          onClick={props.onClick}
+          to={`/projects/${search.original.metadata.project}/dashboards/${search.original.metadata.name}`}
+          key={`dashboard-${search.original.metadata.project}-${search.original.metadata.name}`}
+        >
+          <Box sx={{ display: 'flex' }} flexDirection="row" alignItems="center">
+            <span
+              dangerouslySetInnerHTML={{
+                __html: kvSearch.render(search.original, search.matched, {
+                  pre: '<strong style="color:darkorange">',
+                  post: '</strong>',
+                  escapeHTML: true,
+                }).metadata.name,
+              }}
+            />
+          </Box>
+          <Chip label={`${search.original.metadata.project}`} size="small" variant="outlined" />
+        </Button>
+      ))}
+    </Box>
+  );
+}
+
+export function SearchBar() {
+  const isMobileSize = useIsMobileSize();
+  const [open, setOpen] = React.useState(false);
+  const [query, setQuery] = React.useState('');
+  const handleOpen = () => setOpen(true);
+  const handleClose = () => setOpen(false);
+  const dashboardQueryResult = useDashboardList(undefined, true);
+  const projectsQueryResult = useProjectList();
+  return (
+    <Paper sx={{ flexGrow: 2, maxWidth: isMobileSize ? '50%' : '30%' }}>
+      <Button size="small" fullWidth sx={{ display: 'flex', justifyContent: 'space-between' }} onClick={handleOpen}>
```

Comment:
On mobile, I think we should only show the search icon

Reply:
I'm not that sure about it. 

It's not that bad like that:
<img width="350" alt="image" src="https://github.com/perses/perses/assets/4548045/e271d439-2828-4b35-827e-ac465d3c6ec1">